version: "3.8"
services:
  db:
    image: mysql:8.0.27
    restart: always
    environment:
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
    volumes:
      - .dbdata:/var/lib/mysql
    healthcheck:
      test: mysql ${MYSQL_DATABASE} --user=${MYSQL_USER} --password='${MYSQL_PASSWORD}' --silent --execute "SELECT 1;"
      interval: 30s
      timeout: 10s
      retries: 5   
      
  redis:
    image: redis:alpine
    container_name: redis
    environment:
      - REDIS_HOST=redis
    ports:
      - "6380:6379"
    

  app:
    build: .
    volumes:
      - .:/coin32 
    ports:
      - 8000:8000
    image: app:coin32
    container_name: coin32_container
    command:  python manage.py runserver 0.0.0.0:8000
    depends_on:
      db:
        condition: service_healthy

  # Celery
  celery_worker:
    build:
      context: .
      dockerfile: Dockerfile
    image: django_celery_example_celery_worker
    command: celery -A core.celery worker --concurrency 2 -l info -n main -Q clearing_database
    volumes:
      - .:/coin32
    env_file:
      - ./.env
    depends_on:
      - app
      - redis
      - db

  celery_beat:
    build:
      context: .
      dockerfile: Dockerfile
    image: django_celery_example_celery_beat
    command: celery -A core.celery beat
    volumes:
      - .:/coin32
    env_file:
      - ./.env
    depends_on:
      - app
      - redis
      - db